modules/vaos/app/services/vaos/messages_service.rb

```diff
@@ -31,10 +31,31 @@ def get_messages(request_id)
       end
     end
 
+    def post_message(request_id, request_object_body)
+      with_monitoring do
+        params = VAOS::MessageForm.new(user, request_id, request_object_body).params
+        response = perform(:post, messages_url(request_id), params, headers)
+
+        if response.status == 200
+          {
+            data: OpenStruct.new(response.body),
+            meta: {}
+          }
+        else
+          handle_error(response)
+        end
+      end
+    end
+
     private
 
+    def handle_error(response)
+      key = response.status == 204 ? 'VAOS_204_hack' : nil
```

Comment:
Still trying to figure this out -- you're using VAOS_204_hack to indicate a 400 error? seems like it would be easier to understand to mask it right at the source (here, where the error is handled) and make it clear what code is being returned and why.

Reply:
I know precisely what is being returned, but the http status being returned (204) is not helpful seeing that the request actually did nothing at all. We need to interpret this SUCCESS as a FAILURE. Whether that means a 400 Bad Request or 404 Not Found. Technically it's the parent object (the appointment request) that was not found, and this was identified when sending appointment_request_id as lowerCamelCase instead of the upperCamelCase they require.

it's unlikely we will ever run into this issue, but I felt it was important to capture it in the event that there are other ways that this issue could manifest itself.